tools/nnyeah/nnyeah/Program.cs

```diff
@@ -1,23 +1,74 @@
 using System;
 using System.IO;
+using System.Reflection;
+using Mono.Options;
+using System.Collections.Generic;
 
 #nullable enable
 
 namespace nnyeah {
     class Program {
         static void Main (string [] args)
         {
-            if (args.Length != 2) {
-                Console.Error.WriteLine ("Usage: dotnet nnyeah.dll /path/to/input/file.dll /path/to/output/file.dll");
-                Environment.Exit (1);
-            }
-            using var stm = new FileStream (args [0], FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+            var doHelp = false;
+            string? infile = null, outfile = null;
+            var verbose = false;
+            var suppressWarnings = false;
+            var warnings = new List<string> ();
+            var transforms = new List<string> ();
+
+            var options = new OptionSet () {
+                { "h|?|help", o => doHelp = true },
+                { "i=|input=", f => infile = f },
+                { "o=|output=", f => outfile = f },
+                { "v|verbose", o => verbose = true },
+                { "s|suppress-warnings", o => suppressWarnings = true },
+            };
+
+            var extra = options.Parse (args);
+
+            if (doHelp || infile is null || outfile is null) {
+                PrintOptions (options, Console.Out);
+                Environment.Exit (0);
+			}
+
+            using var stm = new FileStream (infile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
```

Comment:
This is going to throw an exception. It is assuming that infill is present, which is not. Will result in an ugly exception given to the user:
```
File name: '/Users/mandel/Xamarin/xamarin-macios/xamarin-macios/tools/nnyeah/wrong.txt'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at nnyeah.Program.Main(String[] args) in /Users/mandel/Xamarin/xamarin-macios/xamarin-macios/tools/nnyeah/nnyeah/Program.cs:line 35
```